test/cql-pytest/test_tools.py

```diff
@@ -738,3 +739,301 @@ def test_external_dir_autodetect_conf_dir_home_env(self, scylla_path, system_scy
                 ext_sstable,
                 system_scylla_local_reference_dump,
                 env={"SCYLLA_HOME": scylla_home_dir})
+
+
+@pytest.fixture(scope="module")
+def schema1_file():
+    """ Create a schema.cql for the schema1, used in sstable validation tests. """
+    with tempfile.NamedTemporaryFile("w+t") as f:
+        f.write("CREATE TABLE ks.tbl (pk int, ck int, v text, PRIMARY KEY (pk, ck))")
+        f.flush()
+        yield f.name
+
+
+@pytest.fixture(scope="module")
+def schema2_file():
+    """ Create a schema.cql for the schema2, used in sstable validation tests. """
+    with tempfile.NamedTemporaryFile("w+t") as f:
+        f.write("CREATE TABLE ks.tbl (pk int, ck int, v text, s text STATIC, PRIMARY KEY (pk, ck))")
+        f.flush()
+        yield f.name
+
+
+@pytest.fixture(scope="module")
+def large_rows():
+    """ Create a set of large rows, used in the sstable validation tests. """
+    rows = []
+    v_1k = 'v' * 1024
+    for ck in range(128):
+        ck_raw = "0004{:08x}".format(ck)
+        rows.append({ "type": "clustering-row", "key": { "raw": ck_raw }, "columns": { "v": { "is_live": True, "type": "regular", "timestamp": 1680263186359882, "value": v_1k } } })
+    return rows
+
+
+def find_component(generation, component_type, sst_dir):
+    comps = glob.glob(os.path.join(sst_dir, f"*-{str(generation)}-big-{component_type}.db"))
+    assert len(comps) == 1
+    return comps[0]
+
+
+@pytest.fixture(scope="module")
+def sstable_cache(scylla_path):
+    """ Cache creted sstables, used by the sstable validation tests. """
```

Comment:
This cache is a content addressable cache. I pass in a JSON representation of the sstable content I need and if the hash of this content matches that of an sstable that was written before, the sstable is not written again. I was able to shave of many seconds from the test run time with using this, as the same sstable content is used across many tests here.
I don't know if something like this can be done with the fixture system. Maybe if I created a fixture for each possible sstable content, kindof-like a hash. But that seems very tedious.

Reply:
Yes, I thought to have a separate fixture for different sstable contents - sort of like other tests where I have "table1" and "table2" with different schemas, and for each test in the file pick which table is appropriate for this test.

But I guess that "cache" idea works too and I'm not against it. The main downside with it is that as a reviewer, I have no idea how much this cache really saves - do you end up creating 3 different tables? 30? 300? It's not immediately visible.

Reply:
The amount of different sstables created currently is around 10 (that order of magnitude). Some of them are quite large (see the `large_rows()` fixture -- which is used to cache the large rows too).